src/libraries/System.Diagnostics.DiagnosticSource/ref/System.Diagnostics.DiagnosticSourceActivity.cs

```diff
@@ -121,4 +127,49 @@ public abstract partial class DiagnosticSource
         public System.Diagnostics.Activity StartActivity(System.Diagnostics.Activity activity, object? args) { throw null; }
         public void StopActivity(System.Diagnostics.Activity activity, object? args) { }
     }
+    public enum ActivitySourceEventOperation
+    {
+        SourceCreated = 0,
+        ActivityStarted = 1,
+        ActivityStopped = 2
+    }
+    public enum ActivityKind
+    {
+        Internal = 1,
+        Server = 2,
+        Client = 3,
+        Producer = 4,
+        Consumer = 5,
+    }
+    public sealed class ActivitySourceEventArgs : System.EventArgs
+    {
+        private ActivitySourceEventArgs() { throw null; }
+        public System.Diagnostics.ActivitySourceEventOperation Operation { get; }
+    }
+    public readonly struct ActivityContext
+    {
+        public ActivityContext(System.Diagnostics.ActivityTraceId traceId, System.Diagnostics.ActivitySpanId spanId, System.Diagnostics.ActivityTraceFlags traceOptions, string? traceState = null) { throw null; }
+        public System.Diagnostics.ActivityTraceId TraceId { get; }
+        public System.Diagnostics.ActivitySpanId SpanId { get; }
+        public System.Diagnostics.ActivityTraceFlags TraceFlags { get; }
+        public string? TraceState { get; }
+    }
+    public readonly struct ActivityLink
+    {
+        public ActivityLink(System.Diagnostics.ActivityContext context) { throw null; }
+        public ActivityLink(System.Diagnostics.ActivityContext context, System.Collections.Generic.IDictionary<string, object>? attributes) { throw null; }
+        public System.Diagnostics.ActivityContext Context { get; }
+        public System.Collections.Generic.IDictionary<string, object>? Attributes { get; }
+    }
+    public sealed class ActivitySource : IDisposable
+    {
+        private ActivitySource() { throw null; }
+        public ActivitySource(string name) { throw null; }
+        public static event EventHandler<System.Diagnostics.ActivitySourceEventArgs>? OperationEvent { add { } remove { } }
+        public event EventHandler<System.Diagnostics.ActivitySourceEventArgs>? ActivityEvent { add { } remove { } }
+        public static System.Collections.Generic.IEnumerable<System.Diagnostics.ActivitySource> ActiveList => throw null;
+        public string Name { get; }
+        public Activity? CreateActivity(System.Diagnostics.ActivityContext context = default) { throw null; }
```

Comment:
@lmolkova you mentioned we need to have more overloads for this API. could you please list what other parameters should be passed to the overload?

Reply:
this is all possible options (needed/potentially needed for sampling):

https://github.com/open-telemetry/opentelemetry-dotnet/blob/faafd05d3f36f81d4b0f6ea4eaa66e40e3987cc8/src/OpenTelemetry.Api/Trace/Tracer.cs#L70

- name of the Activity - we have it already
- parent context - we have it
- Links - needed for sampling
- Attributes - may be needed for sampling, but looks like it's a future case

This is OpenTelemetry spec on it 
https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-tracing.md#span-creation

Reply:
Thanks. looks we need overload takes the Links and optional start time.

Reply:
Please look at the latest update and let me know if what I did is good enough. Thanks.